src/qibolab/instruments/qm/config/devices.py

```diff
@@ -0,0 +1,97 @@
+from dataclasses import dataclass, field
+
+from qibolab.components.configs import OscillatorConfig
+
+from ..components import OpxOutputConfig, QmAcquisitionConfig
+
+__all__ = [
+    "OctaveOutput",
+    "OctaveInput",
+    "Controller",
+    "Octave",
+]
+
+
+DEFAULT_INPUTS = {"1": {}, "2": {}}
+"""Default controller config section.
+
+Inputs are always registered to avoid issues with automatic mixer
+calibration when using Octaves.
+"""
+
+
+class PortDict(dict):
```

Comment:
Could be a bit misleading, because in principle here we are not directly working with JSONs. It is just that qm-qua complains if I pass a `config` with integers as keys, they need strings everywhere, even for port identifiers. I am not sure if this is because they are internally converting to JSON (very likely) or some other reason.

Reply:
Fine both ways.
You could even call `StrKeyDict`. But we could wait for when this will be used beyond the ports.

*I'm pretty sure it's JSON, also because the docs seem based on OpenAPI (and in general it makes sense). However, it's pretty much irrelevant for us*